### ISSUE TITLE
Code appt lists to reduce screenreader noise - 508 Defect 3

### ISSUE THREAD

**Issue by outerpress** [department-of-veterans-affairs/va.gov-team#99926]
## Background/Request

From the 508 Audit:

> WCAG 1.3.1 Information and Relationships | Information, structure, and relationships conveyed through presentation can be programmatically determined or are available in text.
Upcoming List
12/18/2024: Cew - The appointments list has each list item coded as a one row grid. so as you navigate through the list you have to hear each line item as a grid instead of a list item and this is very confusing and difficult to listen to as you move from appointment to appointment. In addition you can't navigate the grid using table commands. Either the grid should be listed as presentation only so Jaws doesn't read it or use some other control to gain that visual presentation.

All lists use `<div role="grid">` to define the rows. This adds noise when using a screenreader, where it reads "row one of one, column two of two" etc. Example:

https://github.com/user-attachments/assets/192b16c6-4356-4452-8ba3-ee2158453bc8

### Goal

Identify options for recoding the list to reduce noise caused by `role=grid`

### Requirements to Consider

---

## Tasks
- Answer the following questions:
  - Q. What is `role=grid` currently doing in the code? Would removing it solve the problem?
  - Q. If not, what are the options to reduce the noise caused by `role=grid` when navigating through the lists with a screenreader?

## Time Box
_ hours

## Definition of Done
- Tasks are complete
- Research and recommendations are documented in this ticket

---
## How to configure this issue
- [ ] **Attached to an Epic** (what body of work is this a part of?)
- [ ] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)

**Comment by outerpress** [comment]
@JRRoof @simiadebowale this Spike is to address/research an issue brought up by the 508 audit. Let me know if any other info would be helpful. Also fyi @rileyorr

**Comment by rileyorr** [comment]
Thanks for looping me in! I start poking around with my screen reader and inspecting the code to familiarize myself with the problem. It's interesting this is coded as a list with grids rather than a table - the grids are basically acting like nested tables so there's going to be a lot of noise due to that. For instance, as I tab further into the row it announces "table 2" and "table 3" which can be confusing to screen reader users in itself as to the context of what they're interacting with. 

Here's some early thoughts:
- I noticed there's sr-only text reading the date and time for the appointment. The visual text for the time is using `aria-hidden=true` however the date is not. You could add `aria-hidden=true` to the visual date text to hide it and reduce some overall noise.
- You might also consider restructuring the list/table to use less grids and that _may_ help although we'd have to test it. An example @coforma-jamie helped me noodle on might be something like: 

```
<li>
<div role=“grid”>
	<div role=“rowgroup”>
		<div role=“row”>
			<div role=“gridcell”>30 Thur</div>
			<div role=“gridcell”>12:40 pm MT</div>
			<div role=“gridcell”>Primary care<br>At Cheyenne VA….
</li>
```
- Right now this is a bunch of divs and grids that are acting like tables, so it may be beneficial to explore using a semantic HTML table which would be more accessible to screen reader users overall.

**Comment by JunTaoLuo** [comment]
As part of the investigation into this spike we found that it's possible to completely remove the roles so I'm treating this as a engineering ticket instead of a spike with https://github.com/department-of-veterans-affairs/vets-website/pull/34433 as the PR for functional changes intended for production.

**Comment by JunTaoLuo** [comment]
@outerpress this is ready for validation in staging.

**Comment by outerpress** [comment]
@rileyorr could you also take a look at/listen to this one?

**Comment by outerpress** [comment]
@JunTaoLuo this is sounding sooooo much better. Let's wait for Riley's feedback before closing it out, but definitely an improvement

**Comment by rileyorr** [comment]
@outerpress thanks for the ping! I checked and it is definitely sounding way better! However, I noticed there's an issue with the list nesting where the correct number of items in the list is not being read out. I took a recording of what I'm hearing. Essentially the list is reading out x of 4 instead of x of 6 because it's creating the list from the date parent (ex. 20 Feb) rather individual appointments. 

Let me know if my explanation in the video helps or if you have more questions!

https://github.com/user-attachments/assets/53fc9171-6026-43e4-b8e4-38bcc6db4f12

**Comment by outerpress** [comment]
Thanks @rileyorr ! @JunTaoLuo has this worked change how the items are counted? I'm wondering if the issue Riley reported is a separate one that we should handle in another ticket.

**Comment by JunTaoLuo** [comment]
@outerpress No this change wouldn't have made a difference with the structure of the appointments which is currently implemented as a two tier system with dates in the first tier then appointments for each date if needed. I do think that investigating moving to a single tier list for VO but maintaining the display behavior (i.e. one date for all appointments on the same date) will need additional investigation and should be handled in another ticket.

**Comment by outerpress** [comment]
@rileyorr I'll set up a separate ticket to look at the list count issue you noted, will ping you over there once I get it together.

**Comment by rileyorr** [comment]
@outerpress thank you so much! I appreciate your and @JunTaoLuo's collaboration and look forward to working on this issue together 👍
